src/Grpc.Net.Client/Internal/GrpcCall.cs

```diff
@@ -531,14 +531,26 @@ private async Task SendAsync(HttpClient client, HttpRequestMessage request)
                 });
             }
 
-            if (Options.Credentials != null)
+            if (Options.Credentials != null || Channel.CallCredentials?.Count > 0)
             {
                 // In C-Core the call credential auth metadata is only applied if the channel is secure
                 // The equivalent in grpc-dotnet is only applying metadata if HttpClient is using TLS
                 // HttpClient scheme will be HTTP if it is using H2C (HTTP2 without TLS)
                 if (client.BaseAddress.Scheme == Uri.UriSchemeHttps)
                 {
-                    await ReadCredentialMetadata(client, request, Options.Credentials).ConfigureAwait(false);
+                    var configurator = new DefaultCallCredentialsConfigurator();
+
+                    if (Options.Credentials != null)
```

Comment:
if both callCredentials are compositeChannel credentials are provided, will the order of invocation match to Grpc.Core behavior? If we checked that, it might be good to add a comment that explicitly mentions that.

Reply:
I didn't look into it. Do you know the order off the top of your head?

Reply:
I'd expect that the call credentials from composite channel credentials get invoked first, but I don't know for sure, we'd need to try.